### PR TITLE
Support passing cluster-wide user tags

### PR DESCRIPTION
**Pull request by xlucas**
### What does this PR do?

Allow passing extra tags to the cluster-agent in order to be able to define a common set of tags that should be passed on to all cluster resources collected by the orchestrator explorer controller.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -216,3 +216,5 @@ replace k8s.io/klog => k8s.io/klog v1.0.1-0.20200310124935-4ad0115ba9e4
 
 // bitbucket.org/ww/goautoneg has disappeard
 replace bitbucket.org/ww/goautoneg => github.com/munnerz/goautoneg v0.0.0-20190414153302-2ae31c8b6b30
+
+replace github.com/DataDog/agent-payload => github.com/DataDog/agent-payload v4.44.1-0.20201007091830-96613c2a95e4+incompatible
```
Comment (xlucas) [inline go.mod:220]: Note: will bump this to the next release of agent-payload once the PR gets merged.

**Review by mfpierre — APPROVED** [review]
LGTM maybe worth a release note?

**Review by ahmed-mez — APPROVED** [review]

**Review by ahmed-mez — COMMENTED** [review]
any reason why we don't use `DD_TAGS` here ?

**Comment by xlucas** [comment]
@ahmed-mez There are a couple reasons for not using it:
- dd_tags is used to set custom host tags on the agent, whereas this is a cluster agent option that would be used to define custom cluster tags, so from the user perspective it could be confusing to have the same config option with two different behaviours
- this is a relatively new controller and it didn't sound like there's a common use case for setting such tags in other cluster-agent controllers, so I would rather see this namespaced than declared as a "root" config option for that agent.

I think we can reconsider moving it to dd_tags later on if there's another similar need that arises in the future. WDYT?

**Comment by xlucas** [comment]
In the future we'll want to consider that migration I believe but AFAIK we're not actively looking into it and it is hard to anticipate the shape it'll eventually take.

As we need this to fix a few issues I'd personally rather see it kept as is in the meantime as we need this to fix a few collection issues at the moment.

**Pull request merged**